core/src/prelude.rs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! prelude to simplify imports for consumers of `mc_core`
+
+pub use mc_core_types::{
+    account::{Account, PublicSubaddress, RingCtAddress, SpendSubaddress, ViewSubaddress},
```

Comment:
See question above about when to use `Account` as `use mc_core_types::Account` or `use mc_core_types::prelude::Account`.

Reply:
ooh yeah, good catch, given the prelude i probably shouldn't be re-exporting that without a module.

Reply:
right, so this is an exercise in dependency-pickle-avoidance... my concept for `mc_core` is that eventually it'd provide all the basics you need to Do MobileCoin Stuff; so basic types and re-exporting functionality like ring signatures / transactions / memos (while maintaining the no_std and no alloc reqs) for internal and external use. 
(again in the context of hardware support, this is the crate other apps will depend on)

the rules as they stand:
- internal packages **SHOULD** depend on `mc_core_types` _unless_ they require functionality in `mc_core` (`RingCTAddress` or `Subaddress` being two particular examples).
- anything `mc_core` needs to re-export (ie. `mc-crypto-ring-signature` and children) **MUST** depend on `mc_core_types` for the avoidance of dependency loops.

which i'll add to the readme / crate docs

Reply:
I'm not sure this is an answer to the question (I may be misunderstanding), but I'm asking when do you include `prelude` vs when do you omit `prelude` when bringing a type into scope?